integration-cli/docker_cli_ps_test.go

```diff
@@ -540,7 +540,7 @@ func (s *DockerSuite) TestPsListContainersFilterCreated(c *testing.T) {
 	// filter containers by 'create' - note, no -a needed
 	out, _ = dockerCmd(c, "ps", "-q", "-f", "status=created")
 	containerOut := strings.TrimSpace(out)
-	assert.Assert(c, strings.HasPrefix(cID, containerOut))
+	assert.Assert(c, strings.Contains(containerOut, shortCID), "Should have seen '%s' in ps output:\n%s", shortCID, out)
```

Comment:
This test was slightly faulty/weird... The `strings.HasPrefix` args were transposed, so it was checking if the `containerID` has the `ps -q` output as a prefix, which of course only works if the `ps -q` output lists *only* the container we want (in its truncated form). So a parallel run of another test could easily have broken it, for example.

This change is now more consistent with earlier in the same test, and also the other tests in this file which either use the same 12-character truncated ID, or call `docker ps -q --no-trunc` in order to compare the full container ID.

Reply:
Thanks for fixing that; perhaps this was done to prevent any other string in the output (errors?) from causing a false "pass" (only guessing here). I expect that would only occur if the command _failed_, in which case the test would already be marked as failed, so I agree that this approach makes sense.

There's a lot of (some quite old) tests in `integration-cli` that could use some TLC some of them are quite brittle (and indeed running them in parallel, or in some cases, possible leftovers from previous tests, could make them fail).